HW4_Uzun/AAmigo.py

```diff
@@ -0,0 +1,181 @@
+         '- charged': ['E', 'D'], '+ charged': ['K', 'H', 'R']})
+    profile = dict({'hydrophobic': 0.0, 'polar': 0.0, '- charged': 0.0, '+ charged': 0.0})
+
+    for amino_acid in aa_seq:
+        for group_name, group_list in aa_biochemistry.items():
+            if amino_acid in group_list:
+                profile[group_name] += 1
+
+    for group, count in profile.items():
+        profile[group] = round((count/len(seq)), 2)
+    return profile
+
+
+def amino_acid_substring(seq: str):
+    """
+
+    Searches for a substring of amino acids in the entire amino acid sequence.
+    Takes a string of amino acids and a substring, which should be found.
+    Returns the position where the searched one was found for the first time.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    aa_seq = list(seq)
+    aa_seq_upper = []
+    for sequences in aa_seq:
+        upper_case = sequences.upper()
+        aa_seq_upper.append(upper_case)
+    amino_acids = aa_seq_upper[:-1]
+    substring = aa_seq_upper[-1]
+    results = []
+    for sequences in amino_acids:
+        subst = sequences.find(substring)
+        results.append(subst)
+    return results
+
+
+def amino_acid_count(seq: str):
+    """
+
+    Finds how many times a particular sequence(s) occurs in the original one.
+    Takes a string of amino acids and a substring, which should be counted.
+    Returns the count of searched amino acids.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    aa_seq = list(seq)
+    aa_seq_upper = []
+    for sequences in aa_seq:
+        upper_case = sequences.upper()
+        aa_seq_upper.append(upper_case)
+    amino_acids = aa_seq_upper[:-1]
+    substring = aa_seq_upper[-1]
+    results = []
+    for sequences in amino_acids:
+        aa_count = sequences.count(substring)
+        results.append(aa_count)
+    return results
+
+
+def protein_length(*seq: str):
+    """
+
+    Calculate the length (number of amino acids) of a protein.
+    Takes a string of amino acids, returns the number.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    lengths = []
+
+    for sequences in seq:
+        lengths.append(len(sequences))
+
+    return lengths
+
+
+def essential_amino_acids(*seq: str):
+    """
+
+    Calculate the number of essential amino acids based on its amino acids sequence.
+    Takes a string of amino acids, returns only the essential amino acids.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    eaa_dictionary = ['H', 'I', 'K', 'L', 'M', 'F', 'T', 'W', 'V', 'h', 'i', 'k', 'l', 'm', 'f', 't', 'w', 'v']
+    eaa_list = []
+
+    for sequences in seq:
+        eaa_seq = []
+        for amino_acid in sequences:
+            if amino_acid in eaa_dictionary:
+                eaa_seq.append(amino_acid)
+        eaa_list.append(eaa_seq)
+
+    return eaa_list
+
+
+def aa_tools(*args):

```

Comment:
```suggestion
def run_aa_tools(*args):
```
Или как-то по-другому, но чтобы был глагол.